api/client/beacon/client.go

```diff
@@ -159,7 +148,7 @@ func withSSZEncoding() reqOption {
 
 // get is a generic, opinionated GET function to reduce boilerplate amongst the getters in this package.
 func (c *Client) get(ctx context.Context, path string, opts ...reqOption) ([]byte, error) {
-	u := c.urlForPath(path)
+	u := c.baseURL.ResolveReference(&url.URL{Path: path})
```

Comment:
`ResolveReference` used this way basically just adds a path to the base URL, but importantly it first makes a copy of the baseURL and returns the mutated copy, so nothing ever needs to lock or mutate baseURL.
https://pkg.go.dev/net/url#URL.ResolveReference

Reply:
might want to refactor my own to use this.